src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -253,13 +287,15 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
 
                     _bundleHttpContextAccessor.HttpContext = httpContext;
 
+                    _resourceIdProvider.Create = () => persistedId;
```

Comment:
Not a blocker, but since this provider is still ambient in a context, I still feel this should be reverted to the previous state after the invoke